src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -240,6 +273,32 @@ public void httpWriteResponseEnsureOrderTest() throws IOException {
 		httpRequestSender("POST", String.join("", messages), smallStr);
 	}
 	
+	@Test
+	public void httpWriteReceiveAllByteValuesTest() throws IOException {
+		byte[] byteSet = getByteSet();
+		listener.setRequestHandler(context -> {
+			RelayedHttpListenerResponse response = context.getResponse();
+			response.setStatusCode(STATUS_CODE);
+			response.setStatusDescription(STATUS_DESCRIPTION);
+			
+			try {
+				response.getOutputStream().writeAsync(byteSet, 0, byteSet.length).join();
+			} catch (Exception e) {
+				fail(e.getMessage());
```

Comment:
>fail [](start = 4, length = 4)

If this fail occurs in the requestHandler callback does the unit test fail? (Actually, I think so because the correct response most likely won't appear.